pennylane/transforms/defer_measurements.py

```diff
@@ -101,9 +113,305 @@ def null_postprocessing(results):
+            super().__init__()
+            self._aux_wires = Wires(aux_wires)
+
+            # We use a dict here instead of a normal int variable because we want the state to mutate
+            # when we interpret higher-order primitives
+            self.state = {"cur_idx": 0}
+
+        def cleanup(self) -> None:
+            """Perform any final steps after iterating through all equations.
+
+            Blank by default, this method can clean up instance variables. Particularly,
+            this method can be used to deallocate qubits and registers when converting to
+            a Catalyst variant jaxpr.
+            """
+            self.state = {"cur_idx": 0}
+
+        def interpret_dynamic_operation(self, data, struct, idx):
+            """Interpret an operation that uses mid-circuit measurement outcomes as parameters.
+
+            * This will not work if mid-circuit measurement values are used to specify
+              operator wires.
+            * This will not work if more than one parameter uses mid-circuit measurement values.
+
+            Args:
+                data (TensorLike): Flattened data of the operator
+                struct (PyTreeDef): Pytree structure of the operator
+                idx (int): Index of mid-circuit measurement value in ``data``
+
+            Returns:
+                None
+            """
+            mv = data[idx]
+            for branch, value in mv.items():
+                data[idx] = value
+                op = jax.tree_util.tree_unflatten(struct, data)
+                qml.ctrl(op, mv.wires, control_values=branch)
+
+        def interpret_operation(self, op: "pennylane.operation.Operator"):
+            """Interpret a PennyLane operation instance.
+
+            Args:
+                op (Operator): a pennylane operator instance
+
+            Returns:
+                Any
+
+            This method is only called when the operator's output is a dropped variable,
+            so the output will not affect later equations in the circuit.
+
+            See also: :meth:`~.interpret_operation_eqn`.
+
+            """
+            data, struct = jax.tree_util.tree_flatten(op)
+
+            for i, d in enumerate(data):
+                if isinstance(d, MeasurementValue):
+                    return self.interpret_dynamic_operation(data, struct, i)
+
+            return jax.tree_util.tree_unflatten(struct, data)
+
+        def interpret_measurement(self, measurement: "qml.measurement.MeasurementProcess"):
+            """Interpret a measurement process instance.
+
+            Args:
+                measurement (MeasurementProcess): a measurement instance.
+
+            See also :meth:`~.interpret_measurement_eqn`.
+
+            """
+            if measurement.mv is not None:
+                kwargs = {"wires": measurement.wires, "eigvals": measurement.eigvals()}
+                if isinstance(measurement, CountsMP):
+                    kwargs["all_outcomes"] = measurement.all_outcomes
```

Comment:
Note that counts doesn't work with program capture.  So how would we hit this?

Reply:
In the tests, I've hit it with `make_jaxpr` and `CollectOpsAndMeas`. No invalid types are returned unless we actually try to execute AFAIK

Reply:
But I think you are correct, it will fail if we execute.